File: backend/src/main/java/com/whiteblue/backend/domain/application/Application.java
package com.whiteblue.backend.domain.application;

import com.whiteblue.backend.domain.user.User;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@EntityListeners(AuditingEntityListener.class)
@Entity
public class Application {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private String name;

    @NotNull
    private String phoneNumber;

    @NotNull
    private String major;

    @NotNull
    private String introduction;

    @CreatedDate
    private LocalDateTime createdDate;

    @LastModifiedDate
    private LocalDateTime modifiedDate;

    @NotNull
    @OneToOne(cascade = CascadeType.PERSIST, fetch = FetchType.LAZY)
    private User writer;

    @Builder
    public Application(Long id, String name, String phoneNumber, String major, String introduction, User writer) {
        this.id = id;
        this.name = name;
        this.phoneNumber = phoneNumber;
        this.major = major;
        this.introduction = introduction;
        this.writer = writer;
    }
}

File: backend/src/main/kotlin/com/whiteblue/backend/domain/application/Application.kt
package com.whiteblue.backend.domain.application

import com.whiteblue.backend.domain.user.User
import com.whiteblue.backend.util.BaseTimeEntity
import jakarta.persistence.*
import jakarta.validation.constraints.NotNull

@Table(name = "application")
@Entity
class Application(
    @field:NotNull
    var name: String,

    @field:NotNull
    var phoneNumber: String,

    @field:NotNull
    var major: String,

    @field:NotNull
    var introduction: String,

    @field:NotNull
    @OneToOne(cascade = [CascadeType.PERSIST], fetch = FetchType.LAZY)
    var writer: User
) : BaseTimeEntity() {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    val id: Int? = null
}
